nav2_util/include/nav2_util/simple_action_server.hpp

```diff
@@ -34,17 +34,20 @@ class SimpleActionServer
 public:
   typedef std::function<void ()> ExecuteCallback;
 
+  typedef std::function<bool (std::shared_ptr<const typename ActionT::Goal>)> GoalCheckerCallback;
+
   explicit SimpleActionServer(
     typename nodeT::SharedPtr node,
     const std::string & action_name,
     ExecuteCallback execute_callback,
+    GoalCheckerCallback goal_checker_callback = nullptr,
```

Comment:
as an optional argument, all the existing action servers (Planner, ComposePath, BT_Navigator, ...) do not need to bother about it much. But if interested it can be added. I hope this is performant enough?
`Nullptr` seems appropriate for an empty std::function<>..?